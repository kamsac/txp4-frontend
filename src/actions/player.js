import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/player/mock';
import PlayerResource from '../resources/player';

export const RECEIVE_PLAYER = 'player:receivePlayer';
export function receivePlayer(player) {
  return {
    type: RECEIVE_PLAYER,
    payload: { player },
  };
}

function fetchPlayer() {
  const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

  return (dispatch) => {
    resource.getPlayer()
      .then((response) => {
        dispatch(receivePlayer(response.data));
      });
  };
}

export const REQUEST_PLAYER = 'player:requestPlayer';
export function requestPlayer() {
  return (dispatch) => {
    dispatch(fetchPlayer());
  };
}
