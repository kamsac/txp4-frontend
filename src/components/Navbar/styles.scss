@import '../../variables';

.Navbar {
  border-bottom: $border-width solid $black;
}

.Navbar__ContentMargin {
  align-items: center;
  max-width: 1000px;
  margin: 0 auto;
  display: grid;
  grid-area: Navbar;
  grid-gap: $grid-gap-large;
  grid-template-areas: "AppTitle Navigation LoggedUser";
  grid-template-columns: auto 1fr auto;
}

.AppTitle {
  grid-area: AppTitle;
  margin: 0;
}

.Navigation {
  display: grid;
  grid-area: Navigation;
  grid-auto-columns: minmax(100px, max-content);
  grid-auto-flow: column;
  grid-gap: $grid-gap-small;
  height: $navbar-height;
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.Navigation--LoggedUser {
  grid-area: unset;
}

.NavigationItem {
  align-items: stretch;
  display: grid;
  justify-items: stretch;
}

.NavigationItem__anchor {
  align-items: center;
  display: grid;
  justify-items: center;
  text-decoration: none;
  cursor: pointer;
  color: $link-color;

  &:hover {
    color: $link-hover-color;
  }
}

.LoggedUser {
  align-items: center;
  display: grid;
  grid-area: LoggedUser;
  grid-auto-flow: column;
  grid-auto-columns: minmax(100px, max-content);
  grid-gap: $grid-gap-small;
}

.LoggedUser__nick {

}
