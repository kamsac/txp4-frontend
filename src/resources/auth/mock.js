export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
