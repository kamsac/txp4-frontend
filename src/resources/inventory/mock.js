const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
