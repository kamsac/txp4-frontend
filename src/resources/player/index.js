import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const PlayerResource = {
  getPlayer(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}`);
  },

  getInventory(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory`);
  },

  equipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/equip`);
  },

  unequipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/unequip`);
  },
};

export default PlayerResource;
