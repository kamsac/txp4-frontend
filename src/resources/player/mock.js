export const ITEMS = [
  {
    id: 'a1a1a1',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'b2b2b2',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'c3c3c3',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'd4d4d4',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'f6f6f6',
    name: 'Duper engine',
    vendorId: 'aaa',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1.5,
      },
      {
        name: 'acceleration',
        value: 3,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'f7f7f7',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'g8g8g8',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'h9h9h9',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'i0i0i0',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'j1j1j1',
    name: 'Transmission',
    vendorId: 'ccc',
    equipRegion: 'transmission',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.7,
      },
      {
        name: 'acceleration',
        value: 1.5,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
];

export const PLAYER = {
  login: 'lymak',
  nick: '$s$o$a60l$c82y$fa3m$c82a$a60k',
  inventory: {
    items: ITEMS,
  },
};

const PlayerResourceMock = {
  getPlayer() {
    return new Promise((resolve) => {
      const response = {
        data: PLAYER,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  getInventory() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  unequipItem() {
    return new Promise((resolve) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default PlayerResourceMock;
