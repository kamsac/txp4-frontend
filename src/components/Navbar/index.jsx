import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

export default class Navbar extends React.Component {
  constructor(props) {
    super(props);

    this.navigationItems = [
      {
        href: '/inventory',
        label: 'Inventory',
      },
      {
        href: '/scores',
        label: 'Scores',
      },
    ];
  }

  render() {
    const navigationItems = this.navigationItems.map(navigationItem => (
      <li key={navigationItem.label} className="NavigationItem">
        <Link to={navigationItem.href} className="NavigationItem__anchor">
          {navigationItem.label}
        </Link>
      </li>
    ));

    return (
      <div className="Navbar">
        <h1 className="AppTitle">TXP4</h1>
        <ul className="Navigation">
          {navigationItems}
        </ul>
      </div>
    );
  }
}
