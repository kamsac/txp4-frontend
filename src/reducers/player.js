import inventoryReducer from './inventory';
import { RECEIVE_PLAYER } from '../actions/player';
import { EQUIP_ITEM, RECEIVE_INVENTORY, REQUEST_INVENTORY } from '../actions/inventory';

const initialState = {
  login: '',
  nick: '',
  inventory: {
    items: [],
  },
};

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_PLAYER:
      return Object.assign({}, state, action.payload.player);

    case REQUEST_INVENTORY:
    case RECEIVE_INVENTORY:
    case EQUIP_ITEM: {
      const inventory = inventoryReducer(state.inventory, action);
      return Object.assign({}, state, { inventory });
    }
    default:
      return state;
  }
}
