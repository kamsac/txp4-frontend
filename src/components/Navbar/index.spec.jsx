import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './index';
import dotenvConfiguration from '../../dotenv-configuration';
import { PLAYER } from '../../resources/player/mock';

describe('<Navbar>', () => {
  it('should render', () => {
    dotenvConfiguration.API_URL = jest.fn().mockReturnValue('https://localhost:9000/api');
    const component = shallow(<Navbar
      logoutUser={() => null}
      isUserLogged={false}
    />);
    expect(component).toMatchSnapshot();
  });

  it('should add "login mock" button when no api url is defined', () => {
    dotenvConfiguration.API_URL = jest.fn().mockReturnValue('');
    const component = shallow(<Navbar
      logoutUser={() => null}
      isUserLogged={false}
    />);
    expect(component).toMatchSnapshot();
  });

  it('should try to logout user on logout button click', () => {
    const logoutUserSpy = jest.fn();
    const component = shallow(<Navbar
      logoutUser={logoutUserSpy}
      userLogin={PLAYER.login}
      userNick={PLAYER.nick}
      isUserLogged
    />);
    const logoutButton = component.find('.NavigationItem__anchor').filterWhere(wrapper => wrapper.text() === 'Logout');
    logoutButton.simulate('click');
    expect(logoutUserSpy).toBeCalled();
  });
});
