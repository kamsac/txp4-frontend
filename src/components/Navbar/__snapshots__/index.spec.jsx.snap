// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar> should add "login mock" button when no api url is defined 1`] = `
<div
  className="Navbar"
>
  <div
    className="Navbar__ContentMargin"
  >
    <h1
      className="AppTitle"
    >
      TXP4
    </h1>
    <ul
      className="Navigation"
    >
      <li
        className="NavigationItem"
        key="Inventory"
      >
        <Link
          className="NavigationItem__anchor"
          replace={false}
          to="/inventory"
        >
          Inventory
        </Link>
      </li>
      <li
        className="NavigationItem"
        key="Scores"
      >
        <Link
          className="NavigationItem__anchor"
          replace={false}
          to="/scores"
        >
          Scores
        </Link>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Navbar> should render 1`] = `
<div
  className="Navbar"
>
  <div
    className="Navbar__ContentMargin"
  >
    <h1
      className="AppTitle"
    >
      TXP4
    </h1>
    <ul
      className="Navigation"
    >
      <li
        className="NavigationItem"
        key="Inventory"
      >
        <Link
          className="NavigationItem__anchor"
          replace={false}
          to="/inventory"
        >
          Inventory
        </Link>
      </li>
      <li
        className="NavigationItem"
        key="Scores"
      >
        <Link
          className="NavigationItem__anchor"
          replace={false}
          to="/scores"
        >
          Scores
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
